serviceregistry-server/src/main/java/no/difi/meldingsutveksling/serviceregistry/servicerecord/ServiceRecordFactory.java

```diff
@@ -134,58 +132,52 @@ public ServiceRecordFactory(ServiceregistryProperties properties,
         return Optional.of(serviceRecord);
     }
 
-    @SuppressWarnings("squid:S1166") // Suppress Sonar due to no rethrow/log from certificate ex.
+    public Optional<ServiceRecord> createEduErrorRecord(String orgnr) {
+        return Optional.of(ErrorServiceRecord.create(DPO));
+    }
+
+    @HystrixCommand(fallbackMethod = "createDpeInnsynErrorRecord")
     public Optional<ServiceRecord> createDpeInnsynServiceRecord(String orgnr) {
         if (elmaLookupService.identifierHasInnsynskravCapability(NORWAY_PREFIX + orgnr)) {
-            String pemCertificate;
-            try {
-                pemCertificate = lookupPemCertificate(orgnr);
-            } catch (CertificateNotFoundException e) {
-                log.info(MarkerFactory.receiverMarker(orgnr), String.format("Identifier %s found in ELMA with " +
-                        "DPE Innsyn profile, but certificate not found in Virksert.", orgnr));
-                return Optional.empty();
-            }
-            DpeServiceRecord sr = DpeServiceRecord.of(pemCertificate, orgnr, DPE_INNSYN);
-            return Optional.of(sr);
+            String pemCertificate = lookupPemCertificate(orgnr);
+            return Optional.of(DpeServiceRecord.of(pemCertificate, orgnr, DPE_INNSYN));
         }
         return Optional.empty();
     }
 
-    @SuppressWarnings("squid:S1166") // Suppress Sonar due to no rethrow/log from certificate ex.
+    public Optional<ServiceRecord> createDpeInnsynErrorRecord(String orgnr) {
+        return Optional.of(ErrorServiceRecord.create(DPE_INNSYN));
+    }
+
+    @HystrixCommand(fallbackMethod = "createDpeDataErrorRecord")
     public Optional<ServiceRecord> createDpeDataServiceRecord(String orgnr) {
         if (elmaLookupService.identifierHasInnsynDataCapability(NORWAY_PREFIX + orgnr)) {
-            String pemCertificate;
-            try {
-                pemCertificate = lookupPemCertificate(orgnr);
-            } catch (CertificateNotFoundException e) {
-                log.info(MarkerFactory.receiverMarker(orgnr), String.format("Identifier %s found in ELMA with " +
-                        "DPE Data profile, but certificate not found in Virksert.", orgnr));
-                return Optional.empty();
-            }
-            DpeServiceRecord sr = DpeServiceRecord.of(pemCertificate, orgnr, DPE_DATA);
-            return Optional.of(sr);
+            String pemCertificate = lookupPemCertificate(orgnr);
+            return Optional.of(DpeServiceRecord.of(pemCertificate, orgnr, DPE_DATA));
         }
         return Optional.empty();
     }
 
-    @SuppressWarnings("squid:S1166") // Suppress Sonar due to no rethrow/log from certificate ex.
+    public Optional<ServiceRecord> createDpeDataErrorRecord(String orgnr) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "orgnr". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1172)